tests/js/realm-tests.js

```diff
@@ -1609,6 +1609,27 @@ module.exports = {
             });
     },
 
+    testNoMigrationOnSync: function() {
+        if (!global.enableSyncTests) {
+            return;
```

Comment:
should return a `Promise.resolve()` here. although we might have other tests doing that same thing I think we need to fix them as well

Reply:
I've fixed it for this test, and created a ticket to check up on the rest of the tests.